src/askui/telemetry/processors.py

```diff
@@ -0,0 +1,110 @@
+import abc
+
+from datetime import datetime, timezone
+from typing import Any, TypedDict
+
+from pydantic import BaseModel, HttpUrl
+from askui.logger import logger
+from askui.telemetry.context import TelemetryContext
+
+
+class TelemetryProcessor(abc.ABC):
+    @abc.abstractmethod
+    def record_event(
+        self,
+        name: str,
+        attributes: dict[str, Any],
+        context: TelemetryContext,
+    ) -> None: ...
+
+    @abc.abstractmethod
+    def flush(self) -> None: ...
+
+
+class TelemetryEvent(TypedDict):
+    name: str
+    attributes: dict[str, Any]
+    context: TelemetryContext
+    timestamp: datetime
+
+
+class SegmentSettings(BaseModel):
+    api_url: HttpUrl = HttpUrl("https://tracking.askui.com/v1")
+    write_key: str = "Iae4oWbOo509Acu5ZeEb2ihqSpemjnhY"
+
+
+class Segment(TelemetryProcessor):
```

Comment:
Does Segment support proxy?

Reply:
Did not test it but assume this to be true as it uses `requests` under the hood.